hyperdome/client/hyperdome_client.py

```diff
@@ -479,8 +479,13 @@ def closeEvent(self, event):
         """
         When the main window is closed, do some cleanup
         """
-        self.disconnect_chat()
-
+        self.__log.info("main window recieved closeEvent, cleaning up")
+        try:
+            self.disconnect_chat()
+        except AttributeError:
```

Comment:
It would have disconnect chat, but the variables inside disconnect chat will be unbound if there is no active chat - I guess this handling should be inside disconnect chat instead.

Reply:
I resubmitted with a better solution that can be used in other places that should fail gracefully on encountering an unbound variable. I know there's other spots but I'm not sure where.